Adding initial support for oracle incremental timestamps

kakoni opened the pull request:
+ add tests for incremental
+ Changed sqlplus to Oracle SQL Developer Command Line  (SQLcl)

hito4t commented on embulk-input-oracle/src/main/java/org/embulk/input/oracle/getter/OracleTimestampWithTimeZoneIncrementalHandler.java:
```diff
@@ -0,0 +1,67 @@
+import java.sql.SQLException;
+import java.sql.Timestamp;
+
+public class OracleTimestampWithTimeZoneIncrementalHandler extends AbstractIncrementalHandler
```
This class is not used.

hito4t commented on embulk-input-oracle/src/test/java/org/embulk/input/oracle/OracleTests.java:
```diff
@@ -33,8 +32,9 @@ public static void execute(TestingEmbulk embulk, String sql) throws IOException
+
         ProcessBuilder pb = new ProcessBuilder(
-                "SQLPLUS",
+                "sql",
```
Isn't it "sqlplus"?
There is not "sql" in my environment (Windows).

kakoni replied:
I'm proposing that one should switch to SQLcl (SQL Developer Command Line) (http://www.oracle.com/technetwork/developer-tools/sqlcl/overview/index.html), which is the new sqlplus for this century;) + its backwards compatible with sqplus scripts/commandline options.

hito4t replied:
Thank you!
I see.

But with SQLcl, some tests in `BasicTest` class failed in my environment because of file encoding.
They succeeded when I replaced
`pb.environment().put("NLS_LANG", "American_America.UTF8");`
by
`pb.environment().put("JAVA_TOOL_OPTIONS", "-Dfile.encoding=UTF8");`
in `OracleTests` class.

Would you try it in your environment?

kakoni replied:
Tried it and works ok. I think this is an issue when running sqlcl on windows (https://community.oracle.com/thread/3974824)

hito4t replied:
Thank you!

hito4t commented on embulk-input-oracle/src/main/java/org/embulk/input/oracle/getter/OracleColumnGetterFactory.java:
```diff
@@ -0,0 +1,36 @@
+
+    switch (column.getTypeName()) {
+      case "DATE":
+        return new TimestampWithTimeZoneIncrementalHandler(getter);
```
`TimestampWithoutTimeZoneIncrementalHandler` should be used because Oracle `DATE` type and `TIMESTAMP` type don't have timezone.

kakoni commented:
I've 

1. Removed ```OracleTimestampWithTimeZoneIncrementalHandler```. This was in my earlier tests, but it looks like ```TimestampWithoutTimeZoneIncrementalHandler will do just fine````
2. Fixed few tests
3. Added few words to oracle README about how to test (I guess it isn't that obvious)

hito4t commented:
Thank you!
I've merged after fixing encoding problem.
